consensus/src/marshal/store.rs

```diff
@@ -25,8 +25,9 @@ pub trait Certificates: Send + Sync + 'static {
 
     /// Store a finalization certificate, keyed by height and commitment.
     ///
-    /// Implementations must durably sync the write before returning; successful completion
-    /// implies that the certificate is persisted.
+    /// Implementations must:
+    /// - Durably sync the write before returning; successful completion implies that the certificate is persisted.
+    /// - Disallow overwriting an existing finalization at the same height or commitment.
```

Comment:
I think we may want to use a different term than "disallow" here. Archive will just ignore duplicate writes (but won't error) -> AFAICT marshal doesn't do any checking of its own to prevent calling Archive?

Reply:
Updated the comment. Correct that marshal doesn't do any checking of its own; That happens here:

https://github.com/commonwarexyz/monorepo/blob/119e5f6e7541d1addddcc13ef08a7e17506cda33/storage/src/archive/immutable/storage.rs#L235-L238

& here:
https://github.com/commonwarexyz/monorepo/blob/119e5f6e7541d1addddcc13ef08a7e17506cda33/storage/src/archive/prunable/storage.rs#L301-L304

Reply:
(That said, I don't think it'd actually be harmful to allow overwrites here in another implementation of `Certificates`. All that'd happen is you'd have a later finalization for the same view, which would still represent the same chain.)

Reply:
I think that's correct.